### PR TITLE
docker refactorings

### LINKED ISSUE #81: Improve docker setup for dmsg-server
**Feature description**

Running a `dmsg-server` needs to be simplfiied in two respects:

- `dmsg-server` should generate a default config and save that if it is not passed a config
- the docker image for `dmsg-server` should allow to pass a config file for the dmsg-server

### PR DESCRIPTION
**Pull request by alexadhy**
Fixes #81 

 Changes:	
- Generate default config for dmsg-server when config aren't specified
- Refactor docker image creation, added entrypoint for each image. 

How to test this PR:
https://github.com/skycoin/dmsg/blob/5b17a2e0122afdc91a13db46811146d57b2e35ea/docs/dockerized.md

### CODE REVIEW

File: cmd/dmsg-server/commands/root.go [hunk 1]
```diff
@@ -22,12 +26,18 @@ import (
 	"github.com/skycoin/dmsg/servermetrics"
 )
 
+const (
+	defaultDiscoveryURL = "http://dmsg-discovery:9090"
```
Comment (jdknives) [inline cmd/dmsg-server/commands/root.go]: Should probably point it to the production discovery.

File: cmd/dmsg-server/commands/root.go [hunk 2]
```diff
@@ -112,6 +127,31 @@ type Config struct {
...
+func genDefaultConfig() (io.ReadCloser, error) {
+	pk, sk := cipher.GenerateKeyPair()
+
+	cfg := Config{
+		PubKey:        pk,
+		SecKey:        sk,
+		Discovery:     defaultDiscoveryURL,
+		LocalAddress:  fmt.Sprintf("localhost%s", defaultPort),
+		PublicAddress: defaultPort,
+		MaxSessions:   10,
```
Comment (jdknives) [inline cmd/dmsg-server/commands/root.go]: Probably should increase this to 2048.

**Review by jdknives — COMMENTED** [review]
Looks good. Just some small comments.

File: cmd/dmsg-server/commands/root.go (hunk 1 above)
Comment (alexadhy) [reply cmd/dmsg-server/commands/root.go]: yes, will do

**Comment by jdknives** [comment]
Also add some documentation to the README about how to build and run.

**Comment by jdknives** [comment]
I followed the README documentation. When running the dockerized dmsg.Server, the Server actually points at the local discovery (despite documentation saying it points to prod). It is also necessary to stop the server with `docker stop $container_id` because it does not react to `Ctrl+C` which I consider a bit cumbersome.

**Comment by alexadhy** [comment]
@jdknives:

run it without config:

Screen Shot 2021-06-09 at 18 18 12


also addressed `Ctrl-C` concern.

**Review by jdknives — APPROVED** [review]

**Pull request merged**